### PR TITLE
[13.0][MIG] stock_available_mrp: Migration to 13.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 13.0.

Please @chienandalu can you review it?

@Tecnativa TT27873

### CODE REVIEW

**Review by chienandalu — COMMENTED** [review]
Looks like the test don't work

File: stock_available_mrp/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                # We want to respect the rounding factor of the potential_qty
+                # Rounding down as we want to be pesimistic.
+                potential_qty = bom_id.product_uom_id._compute_quantity(
+                    potential_qty,
+                    product.bom_id.product_tmpl_id.uom_id,
+                    rounding_method="DOWN",
+                )
+
+            res[product.id]["potential_qty"] = potential_qty
+            res[product.id]["immediately_usable_qty"] += potential_qty
```
Comment (chienandalu) [inline stock_available_mrp/models/product_product.py]:
```suggestion
                immediately_usable_qty = potential_qty if bom_id.type != "phantom" else 0
            res[product.id]["potential_qty"] = potential_qty
            res[product.id]["immediately_usable_qty"] += immediately_usable_qty
```

File: stock_available_mrp/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                for p in component_products
+            }
+
+        for product in product_with_bom:
+            # Need by product (same product can be in many BOM lines/levels)
+            exploded_components = exploded_boms[product.id]
+            component_needs = product._get_components_needs(exploded_components)
+            if not component_needs:
+                # The BoM has no line we can use
+                potential_qty = 0.0
```
Comment (chienandalu) [inline stock_available_mrp/models/product_product.py]:
```suggestion
                potential_qty = immediately_usable_qty = 0.0
```

**Review by chienandalu — COMMENTED** [review]
Now Odoo computes kit potential quantities (https://github.com/odoo/odoo/blob/13.0/addons/mrp/models/product.py#L104) although it doesn't do it for the regular manufactured products. So we should ignore them when computing the immediately_usable_qty.

**Review by chienandalu — COMMENTED** [review]
(there are further errors though, but those related to multicompany)

**Comment by victoralmau** [comment]
Thanks a lot @chienandalu, changes done.

**Review by chienandalu — APPROVED** [review]
Great! :+1:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**